src/nvim/ui.c

```diff
@@ -235,6 +235,32 @@ int ui_pum_get_height(void)
   return pum_height;
 }
 
+bool ui_pum_get_pos(int* pwidth, int *pheight, int* prow, int* pcol)
```

Comment:
return false early if `!ui_has(kUIPopupmenu)`

Reply:
Or with my suggestion above: return the builtin pum position.

Reply:
This routine scans for all connected UIs, and pum_pos is only turned on if set_bounds satisfies all checks, including `ui_has(kUIPopupmenu)`

Reply:
Right, but then the caller needs to check `pum_external`, just like `pum_get_height` does.

Reply:
Added `pum_get_internal_pos` and adjusted the logic: return internal pum geometry when non of the external UIs report pum geometry.